connect/runtime/src/test/java/org/apache/kafka/connect/util/KafkaBasedLogTest.java

```diff
@@ -114,29 +113,49 @@ public class KafkaBasedLogTest {
     private static final String TP0_VALUE_NEW = "VAL0_NEW";
     private static final String TP1_VALUE_NEW = "VAL1_NEW";
 
-    private Time time = new MockTime();
-    private KafkaBasedLog<String, String> store;
+    private final Time time = new MockTime();
+    private MockedKafkaBasedLog store;
 
     @Mock
-    private Runnable initializer;
+    private Consumer<TopicAdmin> initializer;
     @Mock
     private KafkaProducer<String, String> producer;
-    private MockConsumer<String, String> consumer;
-    @Mock
```

Comment:
I'm curious, why was this change required - https://github.com/apache/kafka/pull/14153/commits/2c7d2e380d02d0f7dc77944ca5775ddaa3540457?

Reply:
`admin` field needs to be mocked only in methods where previously `setupWithAdmin` was used, if we mock it everywhere then some tests don't pass because they expect that `admin` is `null`.

Reply:
Ah yeah, that makes sense, the other methods are expecting the consumer to be used to retrieve the end offsets instead of the admin.